### PR TITLE
feat(cgmodels): Remove

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Change

- Remove cgmodels
- Rename Pipeline class to workflow to better describe the data

### How to prepare for test
- [ ] ssh to hasta.scilifelab.se
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_hermes -t hermes -b rm-cgmodels`

### How to test
- [ ] Do ...

### Expected test outcome
- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

File: cg_hermes/cli/export.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ class OutputFormat(StrEnum):
 @app.command(name="tags")
 def export_tags_cmd(
     output: OutputFormat = typer.Option(OutputFormat.github),
-    pipeline: Pipeline = None,
+    pipeline: Workflow = None,
```
Comment (diitaz93) [inline cg_hermes/cli/export.py:78]: isn't this str?

File: cg_hermes/cli/validate.py [hunk 2]
```diff
@@ -50,26 +50,26 @@ def validate_deliverables(
...
 
 @app.command("tags")
-def validate_tags_cmd(pipeline: Pipeline):
+def validate_tags_cmd(pipeline: Workflow):
     """Validate the tag maps for one of the definitions"""
     LOG.info("Validating %s common tags", pipeline)
     exit_code = 0
 
-    if pipeline == str(Pipeline.MIP_DNA):
+    if pipeline == str(Workflow.MIP_DNA):
```
Comment (diitaz93) [inline cg_hermes/cli/validate.py:58]: not sure if the casting to str is necessary considering that the class is a StrEnum

File: cg_hermes/cli/validate.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 @app.command("deliverables")
 def validate_deliverables(
     infile: Path,
-    pipeline: Pipeline = typer.Option(Pipeline.FLUFFY, help="Specify pipeline"),
+    pipeline: Workflow = typer.Option(Workflow.FLUFFY, help="Specify pipeline"),
```
Comment (diitaz93) [inline cg_hermes/cli/validate.py:29]: str instead of Workflow in type hint

File: cg_hermes/deliverables.py [hunk 4]
```diff
@@ -53,7 +52,7 @@ class Deliverables:
     def __init__(
         self,
         deliverables: dict[str, list[dict[str, str]]],
-        pipeline: Pipeline,
+        pipeline: Workflow,
```
Comment (diitaz93) [inline cg_hermes/deliverables.py:55]: fix type hint

File: cg_hermes/validate.py [hunk 5]
```diff
@@ -14,10 +13,10 @@
 
 def get_deliverables_obj(
     deliverables: dict[str, list[dict[str, str]]],
-    pipeline: Pipeline,
+    pipeline: Workflow,
```
Comment (diitaz93) [inline cg_hermes/validate.py:16]: fix type hint

**Review by diitaz93 — APPROVED** [review]
Great! Hopefully we can archive cgmodels soon. I have doubts about the type hints, I marked all the ones I think they need to be fixed

File: cg_hermes/cli/export.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg_hermes/cli/export.py:78]: This way should lock it to the defined strings in the class. But I will add | None.

**Review by ivadym — APPROVED** [review]
🔥👍

**Comment by henrikstranneheim** [comment]
CLI 👍 
```
θ66° [henrik.stranneheim@hasta:~] [S_base] 3m8s $ hermes validate deliverables --help
Usage: hermes validate deliverables [OPTIONS] INFILE

  Validate a deliverables file.

Arguments:
  INFILE  [required]

Options:
  --pipeline [balsamic|balsamic-qc|balsamic-umi|balsamic-pon|demultiplex|fastq|fluffy|microsalt|mip-dna|mip-rna|rnafusion|rsync|sars-cov-2|spring|taxprofiler]
                                  Specify pipeline  [default: fluffy]
  --analysis-type [tumor_wgs|tumor_normal_wgs|tumor_panel|tumor_normal_panel]
                                  Specify the analysis type
  --help                          Show this message and exit.
θ67° [henrik.stranneheim@hasta:~] [S_base] 7s $ hermes export tags --help
Usage: hermes export tags [OPTIONS]

  Export tag definitions from hermes

Options:
  --output [github|plain]         [default: github]
  --pipeline [balsamic|balsamic-qc|balsamic-umi|balsamic-pon|demultiplex|fastq|fluffy|microsalt|mip-dna|mip-rna|rnafusion|rsync|sars-cov-2|spring|taxprofiler]
  --help                          Show this message and exit.
θ65° [henrik.stranneheim@hasta:~] [S_base] 7s $ hermes export tags --pipeline balsamic
2024-01-18 08:58:34 hasta.scilifelab.se cg_hermes.cli.export[43755] INFO Running export tags for pipeline balsamic
| Balsamic tags                                                  | Mandatory   | HK tags                                 | Used by               |
|----------------------------------------------------------------|-------------|-----------------------------------------|-----------------------|
| balsamic-config                                                | True        | balsamic-config                         | audit, cg             |
| balsamic-report                                                | True        | balsamic-report                         | audit                 |
| balsamic-dag                                                   | True        | balsamic-dag                            | audit                 |
| html, multiqc-html                                             | True        | multiqc-html                            | audit, deliver, scout |
| json, multiqc-json                                             | True        | multiqc-json                            | audit                 |
| qc-metrics-yaml, yaml                                          | True        | qc-metrics                              | audit, cg, vogue      |
| cram, tumor-cram                                               | True        | tumor, cram                             | deliver, scout        |
| cram, tumor-cram-index                                         | True        | tumor, cram-index                       | deliver, scout        |
| cram, normal-cram                                              | False       | normal, cram                            | deliver, scout        |
| normal-cram-index, cram                                        | False       | normal, cram-index                      | deliver, scout        |
| germline-vcf-tumor, snv, vcf-tumor, dnascope                   | True        | germline, vcf-snv-germline-tumor        | deliver               |
| germline-vcf-tumor-index, snv, vcf-tumor, dnascope             | True        | germline, vcf-snv-germline-tumor-index  | deliver               |
| snv, germline-vcf-normal, vcf-normal, dnascope                 | False       | germline, vcf-snv-germline-normal       | deliver               |
| snv, vcf-normal, germline-vcf-normal-index, dnascope           | False       | germline, vcf-snv-germline-normal-index | deliver               |
| vcf-dnascope, genotype-vcf-dnascope                            | False       | dnascope, normal, vcf                   | cg, genotype          |
| genotype-vcf-dnascope-index, vcf-dnascope                      | False       | dnascope, normal, vcf-index             | cg, genotype          |
| vcf-tumor, germline-vcf-tumor, sv, manta-germline              | True        | germline, vcf-sv-germline-tumor         | deliver               |
| germline-vcf-tumor-index, sv, vcf-tumor, manta-germline        | True        | germline, vcf-sv-germline-tumor-index   | deliver               |
| manta-germline, sv, germline-vcf-normal, vcf-normal            | False       | germline, vcf-sv-germline-normal        | deliver               |
| sv, manta-germline, vcf-normal, germline-vcf-normal-index      | False       | germline, vcf-sv-germline-normal-index  | deliver               |
| research-vcf-svdb, vcf-svdb                                    | True        | svdb, vcf-sv                            | deliver               |
| research-vcf-svdb-index, vcf-svdb                              | True        | svdb, vcf-sv-index                      | deliver               |
| vcf-pass-svdb, research-vcf-pass-svdb                          | True        | svdb, vcf-sv-research                   | deliver               |
| vcf-pass-svdb, research-vcf-pass-svdb-index                    | True        | svdb, vcf-sv-research-index             | deliver               |
| vcf-pass-svdb, clinical-vcf-pass-svdb                          | True        | svdb, vcf-sv-clinical                   | deliver, scout        |
| vcf-pass-svdb, clinical-vcf-pass-svdb-index                    | True        | svdb, vcf-sv-clinical-index             | deliver, scout        |
| vcf-tnscope, research-vcf-tnscope                              | False       | tnscope, vcf-snv                        | deliver               |
| research-vcf-tnscope-index, vcf-tnscope                        | False       | tnscope, vcf-snv-index                  | deliver               |
| research-vcf-pass-tnscope, vcf-pass-tnscope, snv               | False       | tnscope, vcf-snv-research               | deliver               |
| vcf-pass-tnscope, snv, research-vcf-pass-tnscope-index         | False       | tnscope, vcf-snv-research-index         | deliver               |
| vcf-pass-tnscope, snv, clinical-vcf-pass-tnscope               | False       | tnscope, vcf-snv-clinical               | deliver, scout        |
| clinical-vcf-pass-tnscope-index, snv, vcf-pass-tnscope         | False       | tnscope, vcf-snv-clinical-index         | deliver, scout        |
| clinical-cnv-report-pdf, cnv-report-pdf                        | False       | cnv-report, visualization               | deliver, scout        |
| ascat-copynumber, clinical-ascat-copynumber                    | False       | ascatngs, metrics                       | deliver               |
| rd-delly, clinical-rd-delly                                    | True        | delly, coverage                         | deliver               |
| cgh-tumor, cnv-somatic-cgh-tumor                               | False       | cnvkit, tumor, vcf2cytosure             | deliver, scout        |
| cgh-normal, cnv-somatic-cgh-normal                             | False       | tiddit, normal, vcf2cytosure            | deliver, scout        |
| vcf-vardict, research-vcf-vardict                              | False       | vardict, vcf-snv                        | deliver               |
| vcf-vardict, research-vcf-vardict-index                        | False       | vardict, vcf-snv-index                  | deliver               |
| research-vcf-pass-vardict, vcf-pass-vardict, snv               | False       | vardict, vcf-snv-research               | deliver               |
| vcf-pass-vardict, snv, research-vcf-pass-vardict-index         | False       | vardict, vcf-snv-research-index         | deliver               |
| clinical-vcf-pass-vardict, vcf-pass-vardict, snv               | False       | vardict, vcf-snv-clinical               | deliver, scout        |
| clinical-vcf-pass-vardict-index, vcf-pass-vardict, snv         | False       | vardict, vcf-snv-clinical-index         | deliver, scout        |
| cns, cnv-cns                                                   | False       | cnvkit, metrics, segments               | deliver               |
| cnr, cnv-cnr                                                   | False       | cnvkit, regions                         | storage               |
| scatter, cnv-scatter                                           | False       | cnvkit, visualization, scatter          | deliver               |
| cnv-diagram, diagram                                           | False       | cnvkit, visualization, diagram          | deliver               |
| cnv-gene-metrics, gene-metrics                                 | False       | cnvkit, metrics, genes                  | deliver               |
| clinical-cov-tumor-tiddit, cov-tumor-tiddit                    | False       | tiddit, tumor, coverage                 | deliver               |
| clinical-cov-normal-tiddit, cov-normal-tiddit                  | False       | tiddit, normal, coverage                | deliver               |
| research-tmb, tmb, snv, vardict                                | False       | research, vardict, tmb                  | storage               |
| tnscope, tmb, snv, research-tmb                                | False       | research, tnscope, tmb                  | storage               |
| cram, umi-tumor-cram                                           | True        | umi-cram, tumor                         | deliver, scout        |
| umi-tumor-cram-index, cram                                     | True        | umi-cram-index, tumor                   | deliver, scout        |
| cram, umi-normal-cram                                          | False       | umi-cram, normal                        | deliver, scout        |
| cram, umi-normal-cram-index                                    | False       | umi-cram-index, normal                  | deliver, scout        |
| vcf-tnscope-umi, research-vcf-tnscope-umi                      | True        | tnscope-umi, vcf-umi-snv                | deliver               |
| vcf-tnscope-umi, research-vcf-tnscope-umi-index                | True        | tnscope-umi, vcf-umi-snv-index          | deliver               |
| research-vcf-pass-tnscope-umi, snv, vcf-pass-tnscope-umi       | True        | tnscope-umi, vcf-umi-snv-research       | deliver               |
| research-vcf-pass-tnscope-umi-index, vcf-pass-tnscope-umi, snv | True        | tnscope-umi, vcf-umi-snv-research-index | deliver               |
| vcf-pass-tnscope-umi, clinical-vcf-pass-tnscope-umi, snv       | True        | tnscope-umi, vcf-umi-snv-clinical       | deliver, scout        |
| vcf-pass-tnscope-umi, snv, clinical-vcf-pass-tnscope-umi-index | True        | tnscope-umi, vcf-umi-snv-clinical-index | deliver, scout        |
| research-tmb, tmb, snv, tnscope-umi                            | True        | research, tnscope-umi, tmb              | storage               |
```

File: cg_hermes/cli/export.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg_hermes/cli/export.py:78]: Turn out Typer currently cannot handle the None syntax - leaving it

File: cg_hermes/cli/validate.py (hunk 3 above)
Comment (henrikstranneheim) [reply cg_hermes/cli/validate.py:30]: Sam es above

File: cg_hermes/cli/validate.py (hunk 2 above)
Comment (henrikstranneheim) [reply cg_hermes/cli/validate.py]: Fixed

**Comment by henrikstranneheim** [comment]
Deployed to production:

```
Log deploy... done.
hermes version: 1.14.6
```

**Pull request merged**